core/src/test/scala/org/apache/spark/sql/delta/perf/OptimizeMetadataOnlyDeltaQuerySuite.scala

```diff
@@ -319,4 +321,16 @@ class OptimizeMetadataOnlyDeltaQuerySuite
       optimizationDisabledQueryPlan
     }
   }
+
+  test(".collect() and .show() both use this optimization") {
+    val collectPlans = DeltaTestUtils.withLogicalPlansCaptured(spark, optimizedPlan = true) {
+      spark.sql(s"SELECT COUNT(*) FROM $testTableName").collect()
+    }
+    assert(collectPlans.collect { case x: LocalRelation => x }.size === 1)
```

Comment:
Please check if we can match with plan (just like the above tests)
```      "Project [scalar-subquery#0 [] AS #0L, scalar-subquery#0 [] AS #1L]\n" +
        ":  :- LocalRelation [none#0L]\n" +
        ":  +- LocalRelation [none#0L]\n" +
        "+- OneRowRelation
```

Reply:
Do you mean to check `val optimizedPlan = queryDf.queryExecution.optimizedPlan.canonicalized.toString()`?
The query is the same for both collect and show: `(s"SELECT COUNT(*) FROM $testTableName")`
So the optimizedPlan is the same.

Reply:
`DeltaTestUtils.withLogicalPlansCaptured` is going to capture plans for Delta logs. Can we check the content in `LocalRelation.data` to make sure we are checking the right plan?